packages/odie-client/src/index.tsx

```diff
@@ -12,16 +13,38 @@ export const ODIE_THUMBS_DOWN_RATING_VALUE = 0;
 export const ODIE_THUMBS_UP_RATING_VALUE = 1;
 
 export const OdieAssistant: React.FC = () => {
-	const { chat, trackEvent, currentUser, isLoadingEnvironment, isLoadingExistingChat } =
-		useOdieAssistantContext();
+	const {
+		chat,
+		trackEvent,
+		currentUser,
+		isLoadingEnvironment,
+		isLoadingExistingChat,
+		isUserElegible,
+		isLoading,
+	} = useOdieAssistantContext();
 	const [ asisstantLoaded, setAssistantLoaded ] = useState( false );
 	const containerRef = useRef< HTMLDivElement >( null );
 	const messagesContainerRef = useRef< HTMLDivElement >( null );
+	const navigate = useNavigate();
 
 	useEffect( () => {
 		trackEvent( 'chatbox_view' );
 	}, [ trackEvent ] );
 
+	useEffect( () => {
+		const { pathname, search } = location;
+		const shouldRedirectHome = ! isUserElegible && ! isLoading;
```

Comment:
That `isLoading` is not related with the isUerElegible, you should be using `isLoadingEnviroment`